### ISSUE TITLE
Failed to receive header: Cannot perform this action while handshake is in progress.

### ISSUE THREAD

**Issue by sakarimov** [GothenburgBitFactory/taskwarrior#2961]
#### Failed to receive header: Cannot perform this action while handshake is in progress.

* What command(s) did you run?
```task sync``` on inthe.am task server
* What did you expect to happen?
it should sync my tasks between local and inthe.am server
* What actually happened?
it gives me this error
```
Failed to receive header: Cannot perform this action while handshake is in progress.
Sync failed.  Could not connect to the Taskserver.
```
* Paste the output of the ```task diag``` command.
```
task 2.6.2
   Platform: Linux

Compiler
    Version: 11.2.0
       Caps: +stdc +stdc_hosted +LP64 +c8 +i32 +l64 +vp64 +time_t64
 Compliance: C++17

Build Features
      CMake: 3.22.3
    libuuid: libuuid + uuid_unparse_lower
  libgnutls: 3.7.3
 Build type:

Configuration
       File: /home/karim/.taskrc (found), 2795 bytes, mode 100644
       Data: /home/karim/.task (found), dir, mode 40755
    Locking: Enabled
         GC: Enabled
    $EDITOR: vim
     Server: inthe.am:53589
         CA: /home/karim/.task/pems/ca.cert.pem, readable, 2405 bytes
Certificate: /home/karim/.task/pems/private.certificate.pem, readable, 2710 bytes
        Key: /home/karim/.task/pems/private.key.pem, readable, 17761 bytes
      Trust: strict
    Ciphers: NORMAL
      Creds: inthe_am/sulthankarimov/************************************

Hooks
     System: Enabled
   Location: /home/karim/.task/hooks
     Active: on-modify.timewarrior    (executable)
   Inactive: on-modify.99-tw-pomodoro (not executable)

Tests
   Terminal: 165x77
       Dups: Scanned 286 tasks for duplicate UUIDs:
             No duplicates found
 Broken ref: Scanned 286 tasks for broken references:
             No broken references found
```

**Comment by manu-cyber** [comment]
I just encountered this issue while replacing my expired certificates (I manage my own server) because I mixed up some variables updating sync/server settings in my client.

You probably have done so already, but maybe it could help double-checking your config with a "fresh set of eyes"? Or just re-downloading the key and certificates and entering the values anew in case you missed a typo or copy&paste error somewhere?
